File: javascript/chapter9/restaurant.js
class Restaurant {
	/**
	 * @param {string} restaurantName
	 * @param {string} cuisineType
	 */
	constructor(restaurantName, cuisineType) {
		this.restaurantName = restaurantName;
		this.cuisineType = cuisineType;
	}

	describeRestaurant() {
		console.log(`Restaurant Name: ${this.restaurantName}`);
		console.log(`Cuisine Type: ${this.cuisineType}`);
	}

	openRestaurant() {
		console.log(`The ${this.restaurantName} Restaurant is now open!`);
	}
}

const restaurant = new Restaurant("Sunny", "Classic");

console.log(restaurant.restaurantName);
console.log(restaurant.cuisineType);

restaurant.openRestaurant();
restaurant.describeRestaurant();

File: python/chapter9/restaurant.py
class Restaurant:
    def __init__(self, restaurant_name: str, cuisine_type: str):
        self.restaurant_name = restaurant_name
        self.cuisine_type = cuisine_type

    def describe_restaurant(self):
        print(f"Restaurant Name: {self.restaurant_name}")
        print(f"Cuisine Type: {self.cuisine_type}")

    def open_restaurant(self):
        print(f"The {self.restaurant_name} Restaurant is now open!")


restaurant = Restaurant("Sunny", "Classic")

print(restaurant.restaurant_name)
print(restaurant.cuisine_type)

restaurant.open_restaurant()
restaurant.describe_restaurant()
